compiler/noirc_evaluator/src/ssa/opt/defunctionalize.rs

```diff
@@ -335,18 +406,45 @@ fn create_apply_functions(
     apply_functions
 }
 
+/// Transforms a [FunctionId] into a [FieldElement]
 fn function_id_to_field(function_id: FunctionId) -> FieldElement {
     (function_id.to_u32() as u128).into()
 }
 
-/// Creates an apply function for the given signature and variants
+/// Creates a single apply function to enable dispatch across multiple function variants
+/// that share the same [Signature] and [RuntimeType].
+///
+/// This function is responsible for generating an entry point that dispatches between several
+/// concrete functions at runtime based on a target field value. It builds a sequence of
+/// conditional checks (if-else chain) to compare the target against each
+/// function's ID, and calls the matching function.
+///
+/// These apply functions are to be aggressively inlined as it is assumed that they will be optimized
+/// away by the constants at the call site.
+///
+/// # Arguments
+/// - `ssa`: A mutable reference to the full [Ssa] structure containing all functions.
+/// - `signature`: The shared [Signature] of all variants.
+/// - `caller_runtime`: The runtime in which the apply function will be called, used to update inlining policies.
+/// - `function_ids`: A non-empty list of [FunctionId]s representing concrete functions to dispatch between.
+///                   This method will panic if `function_ids` is empty.
```

Comment:
```suggestion
/// - `function_ids`: A non-empty list of [FunctionId]s representing concrete functions to dispatch between.
///   This method will panic if `function_ids` is empty.
```

Clippy sees this as overindented